ps-client.js

```diff
@@ -89,13 +90,48 @@
     this.timeout = timeout;
   };
 
+  PredictiveServiceClient.prototype.setSchema = function(cb) { 
+
+    var callback = function(error, resp) {
+      if (resp == null) {
+        cb(error);
+      } else {
+        if (resp.statusCode != 200) {
+          throw new Error("Error connecting to Dato Predictive Service %s." % this.end_point);
+        } else {
+          if (typeof(resp.data) != "string" && 'schema_version' in resp.data) {
+            this.schema = resp.data.schema_version; 
```

Comment:
The more "functional" way of programming here is to return the "schema_version" in callback function instead of setting the value to this directly here, so the caller would look like:

this.getSchema(function(err, schema_version)  {
  if (err) {
    ...
  } else {
    this.schema_version = schema_version
  }
})